File: Kasun/MODIFIED/GoldSpe/GoldSpe/CheckStatus.cs
using Newtonsoft.Json.Linq;
using NLog;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net;
using System.Text;
using System.Threading.Tasks;

namespace GoldSpe
{
    class CheckStatus
    {
        private static Logger logger = LogManager.GetCurrentClassLogger();
        public void Check()
        {
           
           
            RetrieveRobots rt = new RetrieveRobots();
            
            int index = 0;


            List<string> AvailableRobots = rt.GetRobots();


            foreach (string line in AvailableRobots)
            {
               
                index++;
                var Today = DateTime.Now;
                var Yesterday = DateTime.Now.AddDays(-1);
                string x = Today.ToString("yyyy-MM-dd");
                string y = Yesterday.ToString("yyyy-MM-dd");
               

                try
                {
                    string v = string.Format("http://checkmate.ukwest.cloudapp.azure.com:9100/documentServer/documentRepository/documents?query=SourceUrl:" + "\"" + "{0}" + "\"" + " AND PublishDate:" + "[" + y + " TO " + x + "]&sortByField=PublishDate&sortDescending=true", line.Trim());  //Adding Date


                    var response2 = new WebClient().DownloadString(v);




                    JObject jObj1 = JObject.Parse(response2);

                    if (jObj1["Documents"] != null && jObj1["Documents"].HasValues == true)
                    {
                        string id = (string)jObj1["Documents"][0]["_id"];

                        if (response2 != null && string.IsNullOrEmpty(id) == false)
                        {



                            logger.Info("The url of Robot= " + line + "=====is  working\n");
                        }
                        else
                        {
                       
                            logger.Error("The url of Robot= " + line + "=====is not working due to no Robot_ID values \n");
                        }
                    }
                    else
                    {

                  
                        logger.Error("The url of Robot= " + line + "=====is not working due to no Document values \n");
                    }


                }
                catch (Exception e)
                {
                    logger.Error(e);
                }


            }


        }

        

    }
}

File: Kasun/JAVA/kasun/src/main/java/robo/CheckStatus.java
package robo;

import java.util.logging.Logger;
import java.text.DateFormat;
import java.text.SimpleDateFormat;
import java.util.Calendar;
import java.util.Date;
import java.util.List;
import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;
import org.json.simple.parser.ParseException;
import com.sun.jersey.api.client.Client;
import com.sun.jersey.api.client.ClientResponse;
import com.sun.jersey.api.client.WebResource;

public class CheckStatus {    ///// Check Daily status of robots (Starting from here)
	final static Logger logger = Logger.getLogger("CheckStatus.class");

	public void Check() throws JSONException, ParseException {

		RetrieveRobots rtv = new RetrieveRobots();
		List<String> Robo = rtv.getRobots();//// List of available robots in server

		for (String line : Robo) {

			DateFormat dateFormat = new SimpleDateFormat("yyyy/MM/dd HH:mm:ss");
			Date date = new Date();
			String today = dateFormat.format(date).substring(0, 10).replaceAll("[/]", "-"); ////SET DATE TO TODAY
			String yesterday = dateFormat.format(yesterday()).substring(0, 10).replaceAll("[/]", "-");////SET DATE TO YESTERDAY

			try {

				Client client = Client.create();

				WebResource webResource1 = client.resource(
						"http://checkmate.ukwest.cloudapp.azure.com:9100/documentServer/documentRepository/documents?query=SourceUrl%3A%22"
								+ line + "%22PublishDate%3A%5B" + yesterday + "+TO+" + today
								+ "%5D&sortDescending=true");

				ClientResponse response = webResource1.accept("application/json").get(ClientResponse.class);

				if (response.getStatus() != 200) {
					throw new RuntimeException("Failed : HTTP error code : " + response.getStatus());

				}

				String output = response.getEntity(String.class);

				JSONObject jsonObj = new JSONObject(output);

				JSONArray DocumentArray = jsonObj.getJSONArray("Documents");

				JSONObject firstDocument = DocumentArray.getJSONObject(0);///check whether that Data has Text object of Document JSON array 
				String contentText = firstDocument.getString("Text");

				if (contentText != null) {

					logger.info("This          " + line + "================Robot is working");

				} else {
					logger.warning("This             " + line + "================Robot is NOT working");

				}

			} catch (Exception e) {
			}

		}
	}

	private Date yesterday() {
		final Calendar cal = Calendar.getInstance(); //// set the Yesterday date
		cal.add(Calendar.DATE, -1);
		return cal.getTime();
	}

}